File: app/src/main/java/com/ajithvgiri/searchdialog/MainActivity.java
package com.ajithvgiri.searchdialog;

import android.os.Bundle;
import android.support.design.widget.TextInputEditText;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {


    SearchableDialog searchableDialog;
    List<SearchListItem> searchListItems = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        for (int i = 0; i < 10; i++) {
            SearchListItem searchListItem = new SearchListItem(i, "Country " + i);
            searchListItems.add(searchListItem);
        }


        final TextView countryNameTextView = findViewById(R.id.countryNameTextView);
        final TextView countryCodeTextView = findViewById(R.id.countryCodeTextView);

        searchableDialog = new SearchableDialog(this, searchListItems, getString(R.string.country));
        searchableDialog.setOnItemSelected(new OnSearchItemSelected() {
            @Override
            public void onClick(int position, SearchListItem searchListItem) {
                countryCodeTextView.setText(String.valueOf(searchListItem.getId()));
                countryNameTextView.setText(searchListItem.getTitle());
            }
        });
        TextInputEditText countryTextInputEditText = findViewById(R.id.countryTextInputEditText);
        countryTextInputEditText.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                searchableDialog.show();
            }
        });
    }
}

File: app/src/main/java/com/ajithvgiri/searchdialog/MainActivity.kt
package com.ajithvgiri.searchdialog

import android.os.Bundle
import com.google.android.material.textfield.TextInputEditText
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView


class MainActivity : AppCompatActivity() {

    lateinit var searchableDialog: SearchableDialog
    var searchListItems: MutableList<SearchListItem> = ArrayList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        for (i in 0..9) {
            val searchListItem = SearchListItem(i, "Country $i")
            searchListItems.add(searchListItem)
        }
        val countryNameTextView = findViewById<TextView>(R.id.countryNameTextView)
        val countryCodeTextView = findViewById<TextView>(R.id.countryCodeTextView)
        searchableDialog = SearchableDialog(this, searchListItems, getString(R.string.country))
        searchableDialog.setOnItemSelected { _, searchListItem ->
            countryCodeTextView.text = searchListItem.getId().toString()
            countryNameTextView.text = searchListItem.getTitle()
        }
        val countryTextInputEditText = findViewById<TextInputEditText>(R.id.countryTextInputEditText)
        countryTextInputEditText.setOnClickListener { searchableDialog.show() }


    }
}